src/client/connection.rs

```diff
@@ -468,7 +389,7 @@ impl Stream for Connection {
     }
 }
 
-impl AsyncRead for Connection {
+impl<S: futures::AsyncRead + futures::AsyncWrite + Unpin + Send> futures::AsyncRead for Connection<S> {
```

Comment:
You might want to use futures::{AsyncRead, AsyncWrite} to save valuable horizontal space.